File: Customer/src/main/java/com/postman/alt/service/CommentService.java
package com.postman.alt.service;

import com.postman.alt.service.dto.CommentCreateRequest;
import com.postman.alt.service.dto.CommentResponse;
import com.postman.alt.service.dto.CommentUpdateRequest;

import java.util.List;
import java.util.UUID;

public interface CommentService {

    List<CommentResponse> list(UUID workItemId, UUID requesterId);

    CommentResponse create(UUID workItemId, UUID requesterId, CommentCreateRequest request);

    CommentResponse update(UUID commentId, UUID requesterId, CommentUpdateRequest request);

    void delete(UUID commentId, UUID requesterId);
}

File: client/src/app/core/services/comment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { API_BASE_URL } from '../config/api.config';
import { CommentCreateRequest, CommentResponse, CommentUpdateRequest } from '../models/comment.model';

const BASE = `${API_BASE_URL}/api`;

@Injectable({ providedIn: 'root' })
export class CommentService {
  private readonly http = inject(HttpClient);

  list(workItemId: string): Observable<CommentResponse[]> {
    return this.http.get<CommentResponse[]>(`${BASE}/work-items/${workItemId}/comments`);
  }

  create(workItemId: string, request: CommentCreateRequest): Observable<CommentResponse> {
    return this.http.post<CommentResponse>(`${BASE}/work-items/${workItemId}/comments`, request);
  }

  update(commentId: string, request: CommentUpdateRequest): Observable<CommentResponse> {
    return this.http.patch<CommentResponse>(`${BASE}/comments/${commentId}`, request);
  }

  delete(commentId: string): Observable<void> {
    return this.http.delete<void>(`${BASE}/comments/${commentId}`);
  }
}
